openrtx/src/ui/default/ui.c

```diff
@@ -1900,6 +1929,41 @@ void ui_updateFSM(bool *sync_rtx)
                     _ui_menuBack(MENU_TOP);
                 break;
 #endif
+#ifdef CONFIG_SPECTRUM
+            // Spectrum menu screen
+            case MENU_SPECTRUM:
+            // Go to the Spectrum Settings if we press ENTER
+                if(msg.keys & KEY_ENTER)
+                {
+                    display_init();
+                    state.rtxStatus = RTX_RX;
+                    state.ui_screen = SETTINGS_SPECTRUM;
+                }
+                if(msg.keys & KEY_ESC) {
+                    state.rtxStatus = RTX_RX;
+                    radio_enableRx();
```

Comment:
Don't call the radio API outside the rtx thread, please.
You can trigger a radio reconfiguration by setting `sync_rtx = true`